internal/cli/commons.go

```diff
@@ -164,3 +163,25 @@ func matchingCatalogFinder(cmd *cobra.Command, args []string, toComplete string)
 
 	return matches, cobra.ShellCompDirectiveNoFileComp
 }
+
+// filteredMatchingFinder will use the finder func to find the resources with the prefix name
+// It will then filter the matches removing the provided args
+func filteredMatchingFinder(args []string, prefix string, finder func(prefix string) []string) []string {
+	// map to check for already selected applications
+	alreadyMatchedApps := map[string]struct{}{}
```

Comment:
```suggestion
	alreadyMatched := map[string]struct{}{}
```

Given that this is a generic finder `App` makes no sense.

Reply:
This change will induce updating the other places.